plugins/woocommerce/client/blocks/assets/js/blocks/add-to-cart-with-options/variation-selector/attribute-options/frontend.ts

```diff
@@ -1,9 +1,185 @@
 /**
  * External dependencies
  */
-import { store } from '@wordpress/interactivity';
+import type { ChangeEvent } from 'react';
+import { store, getContext, getElement } from '@wordpress/interactivity';
 
-store( 'woocommerce/add-to-cart-with-options', {
-	state: {},
-	actions: {},
-} );
+type Option = {
+	value: string;
+	label: string;
+	isSelected: boolean;
+};
+
+type Context = {
+	name: string;
+	selected: string | null;
+	option: Option;
+	options: Option[];
+};
+
+type PillsContext = Context & {
+	tabIndex?: number;
+	selected?: string;
+	focused?: string;
+};
+
+const { actions: wooAddToCartWithOptions } = store(
+	'woocommerce/add-to-cart-with-options',
+	{},
+	{
+		// Stores are locked to prevent 3PD usage until the API is stable.
+		lock: 'I acknowledge that using a private store means my plugin will inevitably break on the next store release.',
+	}
+);
+
+function setAttribute( name: string, value: string ) {
```

Comment:
The type should be `( name: string, value?: string )`

Reply:
Yes it could be, but the value is never undefined but can be empty string.